sdk/src/client/api/block_builder/input_selection/requirement/amount.rs

```diff
@@ -50,7 +50,7 @@ impl InputSelection {
             }
         }
 
-        for output in self.all_outputs() {
+        for output in self.non_remainder_outputs() {
```

Comment:
why?

Reply:
because this runs in remainder and it doesn't need to look at them, and it breaks if it looks at them in the wrong place